resources/functionMap.php

```diff
@@ -4649,28 +4649,28 @@
 'imagexbm' => ['bool', 'im'=>'resource', 'filename='=>'string|resource|null', 'foreground='=>'int'],
 'Imagick::__construct' => ['void', 'files='=>''],
 'Imagick::__toString' => ['string'],
-'Imagick::adaptiveBlurImage' => ['bool', 'radius'=>'float', 'sigma'=>'float', 'channel='=>'int'],
+'Imagick::adaptiveBlurImage' => ['bool', 'radius'=>'float', 'sigma'=>'float', 'channel='=>'Imagick::CHANNEL_*'],
 'Imagick::adaptiveResizeImage' => ['bool', 'columns'=>'int', 'rows'=>'int', 'bestfit='=>'bool'],
-'Imagick::adaptiveSharpenImage' => ['bool', 'radius'=>'float', 'sigma'=>'float', 'channel='=>'int'],
+'Imagick::adaptiveSharpenImage' => ['bool', 'radius'=>'float', 'sigma'=>'float', 'channel='=>'Imagick::CHANNEL_*'],
```

Comment:
I'm not a big fan of narrowing down parameter types. It requires people to suddently start typehinting PHPStan-specific types in their PHPDocs instead of `int`. For the same reason I haven't merged https://github.com/phpstan/phpstan-src/pull/2163 and https://github.com/phpstan/phpstan-src/pull/2271 yet.

But I think it'd be fine to introduce these typehints in bleedingEdge/PHPStan 2.0.

I've added support for bleedingEdge-specific functionMap: https://github.com/phpstan/phpstan-src/commit/06b746d8e72cc0843707896ec161559bb6a81137

My suggestion is:

1) Submit a PR that only changes return types for everyone.
2) Submit a next PR that changes parameter types in `functionMap_bleedingEdge.php`.

Thank you!

Reply:
> I'm not a big fan of narrowing down parameter types.

I agree with your policy because it makes sense. Thank you for your review!